### PR TITLE
Refactor assertTrue and assertFalse to assertThat

### PR DESCRIPTION
**Pull request by mbaechler**
this is part of a larger transition where I'd like to use only assertj for assertions

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [INFO] -------------------------------------------------------------
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [ERROR] COMPILATION ERROR : 
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [INFO] -------------------------------------------------------------
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [ERROR] /james-project/server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SpamAssassinHandlerTest.java:[21,44] package org.apache.james.util.scanner does not exist
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [ERROR] /james-project/server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SpamAssassinHandlerTest.java:[21,1] static import only from classes and interfaces
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [ERROR] /james-project/server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SpamAssassinHandlerTest.java:[125,44] cannot find symbol
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba]   symbol:   variable STATUS_MAIL_ATTRIBUTE_NAME
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba]   location: class org.apache.james.smtpserver.SpamAssassinHandlerTest
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [ERROR] /james-project/server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SpamAssassinHandlerTest.java:[142,44] cannot find symbol
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba]   symbol:   variable STATUS_MAIL_ATTRIBUTE_NAME
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba]   location: class org.apache.james.smtpserver.SpamAssassinHandlerTest
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [ERROR] /james-project/server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SpamAssassinHandlerTest.java:[158,44] cannot find symbol
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba]   symbol:   variable STATUS_MAIL_ATTRIBUTE_NAME
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba]   location: class org.apache.james.smtpserver.SpamAssassinHandlerTest
[d028971e28b921b0bcc6974f41c4dd6230cdd5ba] [INFO] 5 errors 
```

**Comment by aduprat** [comment]
merged

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void testHashCode() throws AddressException {
     public void testMailAddressString() throws AddressException {
 
         MailAddress a = new MailAddress(GOOD_ADDRESS);
-        Assert.assertTrue(GOOD_ADDRESS.equals(a.toString()));
+        assertThat(GOOD_ADDRESS.equals(a.toString())).isTrue();
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: should be isEqualTo()

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 2]
```diff
@@ -126,13 +126,13 @@ public void testMailAddressStringString() {
         try {
             new MailAddress("local-part", "domain");
         } catch (AddressException e) {
-            Assert.assertTrue(e.getMessage(), false);
+            assertThat(false).describedAs(e.getMessage()).isTrue();
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: this is bad, should be a fail

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 3]
```diff
@@ -126,13 +126,13 @@ public void testMailAddressStringString() {
         try {
             new MailAddress("local-part", "domain");
         } catch (AddressException e) {
-            Assert.assertTrue(e.getMessage(), false);
+            assertThat(false).describedAs(e.getMessage()).isTrue();
         }
         try {
             MailAddress a = new MailAddress("local-part", "-domain");
-            Assert.assertFalse(a.toString(), true);
+            assertThat(true).describedAs(a.toString()).isFalse();
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: idem

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 4]
```diff
@@ -126,13 +126,13 @@ public void testMailAddressStringString() {
...
-            Assert.assertTrue(e.getMessage(), false);
+            assertThat(false).describedAs(e.getMessage()).isTrue();
         }
         try {
             MailAddress a = new MailAddress("local-part", "-domain");
-            Assert.assertFalse(a.toString(), true);
+            assertThat(true).describedAs(a.toString()).isFalse();
         } catch (AddressException e) {
-            Assert.assertTrue(true);
+            assertThat(true).isTrue();
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: useless

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 5]
```diff
@@ -146,7 +146,7 @@ public void testMailAddressInternetAddress() {
             new MailAddress(new InternetAddress(GOOD_QUOTED_LOCAL_PART));
         } catch (AddressException e) {
             System.out.println("AddressException" + e.getMessage());
-            Assert.assertTrue(e.getMessage(), false);
+            assertThat(false).describedAs(e.getMessage()).isTrue();
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: idem

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 6]
```diff
@@ -146,7 +146,7 @@ public void testMailAddressInternetAddress() {
             new MailAddress(new InternetAddress(GOOD_QUOTED_LOCAL_PART));
         } catch (AddressException e) {
             System.out.println("AddressException" + e.getMessage());
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: still a remaining syso?

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 7]
```diff
@@ -158,10 +158,10 @@ public void testGetDomain() {
 
         try {
             MailAddress a = new MailAddress(new InternetAddress(GOOD_ADDRESS));
-            Assert.assertTrue(a.getDomain() + " != " + GOOD_DOMAIN, a.getDomain().equals(GOOD_DOMAIN));
+            assertThat(a.getDomain().equals(GOOD_DOMAIN)).describedAs(a.getDomain() + " != " + GOOD_DOMAIN).isTrue();
         } catch (AddressException e) {
             System.out.println("AddressException" + e.getMessage());
-            Assert.assertTrue(e.getMessage(), false);
+            assertThat(false).describedAs(e.getMessage()).isTrue();
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: so much comment like this...

**Pull request closed without merging**